File: server_akka_scala/src/main/scala/info/rori/lunchbox/server/akka/scala/domain/model/LunchOffer.scala
package info.rori.lunchbox.server.akka.scala.domain.model

import org.joda.money.Money
import org.joda.time.LocalDate


case class LunchOffer(
                       id: Id,
                       name: String,
                       day: LocalDate,
                       price: Money,
                       provider: LunchProviderId
                       ) extends Ordered[LunchOffer] {

  import scala.math.Ordered.orderingToOrdered
  def compare(that: LunchOffer): Int = (this.provider, this.price) compare(that.provider, that.price)
}

File: backend-spring-kotlin/src/main/kotlin/lunchbox/domain/models/LunchOffer.kt
package lunchbox.domain.models

import org.joda.money.Money
import java.time.LocalDate

typealias LunchOfferId = Long

/**
 * Mittagsangebot
 */
data class LunchOffer(
  val id: LunchOfferId,
  val name: String,
  val description: String,
  val day: LocalDate,
  val price: Money?,
  val tags: Set<String>,
  val provider: LunchProviderId,
)
